Shell/LineEditor.cpp

```diff
@@ -123,11 +129,13 @@ void LineEditor::tab_complete_first_token(const String& token)
     bool seen_others = false;
     int index = match - m_path.data();
```

Comment:
Our loops do not iterate over the element returned by `binary_search` (at `index`), because `m_path[index]` is the same string as `completion`, so `cut_mismatching_chars()` wouldn't do anything. Notice we are using `int i = index - 1` and `int i = index + 1`. This means we are missing an element in our suggestions.
We need something like this:
`suggestions.append(m_path[index]);`

Reply:
Hrrm strange, I haven't noticed any missing suggestions yet (comparing with `ls`). I'll add that in regardless